File: master/core-python/print/mainP3.py
"""
Main File Print 3
sidb95
bhatoresiddharth@gmail.com
12 July 2024
"""

from print import Print

class Solution:
  def __init__(self):
    pass

  def mainP3Aux(self, s: str, n: int, i: int) -> None:
    P1 = Print()
    if (i == n):
      P1.printCharacter('\n')
      return None
    else:
      P1.printCharacter(s[i])
    self.mainP3Aux(self, s, n, i +1)
    return None

  def mainP3(self, s: str) -> None:
    self.mainP3Aux(self, s, len(s), 0)
    return None

"""
Main Function 3
"""
def __main__():
  pass

File: master/core-cpp/print/mainP3.cpp
/*
Main File Print 3
sidb95
bhatoresiddharth@gmail.com 
05, 10 July 2024
*/

#include "print.h"

class Solution {
public:
	void mainP3Aux(string strA, int n, int i) {
		Print P1;
		if (i == n) {
			return;
		}
		P1.printStr(strA[i]);
		return;
	}

	void mainP3(string strA) {
		int size1 = strA.size();
		mainP3Aux(strA, size1, 0);
		return;
	}
};
